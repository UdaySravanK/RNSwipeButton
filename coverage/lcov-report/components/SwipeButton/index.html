
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components/SwipeButton</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> components/SwipeButton</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">78.57% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>33/42</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">78.18% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>43/55</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">77.77% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>7/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.48% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>33/41</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="78.04" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.04" class="pct medium">78.04%</td>
	<td data-value="41" class="abs medium">32/41</td>
	<td data-value="78.18" class="pct medium">78.18%</td>
	<td data-value="55" class="abs medium">43/55</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="40" class="abs high">32/40</td>
	</tr>

<tr>
	<td class="file high" data-value="styles.js"><a href="styles.js.html">styles.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-01-24T02:17:55.689Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    