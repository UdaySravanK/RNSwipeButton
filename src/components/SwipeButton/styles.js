import { StyleSheet } from 'react-native';

const Styles = StyleSheet.create({
  container: {
    borderRadius: 100 / 2,
    borderWidth: 1,
    justifyContent: 'center',
    margin: 5,
  },
  title: {
    alignSelf: 'center',
    position: 'absolute',
  },
});

export default Styles;
