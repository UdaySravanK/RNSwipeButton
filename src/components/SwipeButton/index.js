import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Text, View, AccessibilityInfo } from 'react-native';

// Components
import SwipeThumb from '../../components/SwipeThumb';

// Styles
import styles from './styles';

// Constants
import {
  DISABLED_RAIL_BACKGROUND_COLOR,
  DISABLED_THUMB_ICON_BACKGROUND_COLOR,
  DISABLED_THUMB_ICON_BORDER_COLOR,
  RAIL_BACKGROUND_COLOR,
  RAIL_BORDER_COLOR,
  RAIL_FILL_BACKGROUND_COLOR,
  RAIL_FILL_BORDER_COLOR,
  SWIPE_SUCCESS_THRESHOLD,
  THUMB_ICON_BACKGROUND_COLOR,
  THUMB_ICON_BORDER_COLOR,
  TITLE_COLOR,
} from '../../constants';

const SwipeButton = props => {
  const [layoutWidth, setLayoutWidth] = useState(0);
  const [screenReaderEnabled, setScreenReaderEnabled] = useState(false);
  const [isUnmounting, setIsUnmounting] = useState(false);

  /**
   * Retrieve layoutWidth to set maximum swipeable area.
   * Correct layout width will be received only after first render but we need it before render.
   * So render SwipeThumb only if layoutWidth > 0
   */
  const onLayoutContainer = async e => {
    if (isUnmounting || layoutWidth) {
      return;
    }
    setLayoutWidth(e.nativeEvent.layout.width);
  };

  useEffect(() => {
    const handleScreenReaderToggled = isEnabled => {
      if (isUnmounting || screenReaderEnabled === isEnabled) {
        return;
      }
      setScreenReaderEnabled(isEnabled);
    };
    setIsUnmounting(false);
    AccessibilityInfo.addEventListener('change', handleScreenReaderToggled);

    AccessibilityInfo.isScreenReaderEnabled().then(isEnabled => {
      if (isUnmounting) {
        return;
      }
      setScreenReaderEnabled(isEnabled);
    });

    return () => {
      setIsUnmounting(true);
      AccessibilityInfo.removeEventListener(
        'change',
        handleScreenReaderToggled,
      );
    };
  }, [isUnmounting, screenReaderEnabled]);

  const {
    containerStyles,
    disabled,
    disabledRailBackgroundColor,
    disabledThumbIconBackgroundColor,
    disabledThumbIconBorderColor,
    enableRightToLeftSwipe,
    forceReset,
    height,
    onSwipeFail,
    onSwipeStart,
    onSwipeSuccess,
    railBackgroundColor,
    railBorderColor,
    railFillBackgroundColor,
    railFillBorderColor,
    railStyles,
    resetAfterSuccessAnimDelay,
    resetAfterSuccessAnimDuration,
    shouldResetAfterSuccess,
    swipeSuccessThreshold,
    thumbIconBackgroundColor,
    thumbIconBorderColor,
    thumbIconComponent,
    thumbIconImageSource,
    thumbIconStyles,
    title,
    titleColor,
    titleFontSize,
    titleStyles,
    width,
  } = props;
  return (
    <View
      style={[
        styles.container,
        {
          ...containerStyles,
          backgroundColor: disabled
            ? disabledRailBackgroundColor
            : railBackgroundColor,
          borderColor: railBorderColor,
          ...(width ? { width } : {}),
        },
      ]}
      onLayout={onLayoutContainer}>
      <Text
        importantForAccessibility={
          screenReaderEnabled ? 'no-hide-descendants' : ''
        }
        style={[
          styles.title,
          {
            color: titleColor,
            fontSize: titleFontSize,
            ...titleStyles,
          },
        ]}>
        {title}
      </Text>
      {layoutWidth > 0 && (
        <SwipeThumb
          disabled={disabled}
          disabledThumbIconBackgroundColor={disabledThumbIconBackgroundColor}
          disabledThumbIconBorderColor={disabledThumbIconBorderColor}
          enableRightToLeftSwipe={enableRightToLeftSwipe}
          forceReset={forceReset}
          iconSize={height}
          layoutWidth={layoutWidth}
          onSwipeFail={onSwipeFail}
          onSwipeStart={onSwipeStart}
          onSwipeSuccess={onSwipeSuccess}
          railFillBackgroundColor={railFillBackgroundColor}
          railFillBorderColor={railFillBorderColor}
          railStyles={railStyles}
          resetAfterSuccessAnimDelay={resetAfterSuccessAnimDelay}
          resetAfterSuccessAnimDuration={resetAfterSuccessAnimDuration}
          screenReaderEnabled={screenReaderEnabled}
          shouldResetAfterSuccess={shouldResetAfterSuccess}
          swipeSuccessThreshold={swipeSuccessThreshold}
          thumbIconBackgroundColor={thumbIconBackgroundColor}
          thumbIconBorderColor={thumbIconBorderColor}
          thumbIconComponent={thumbIconComponent}
          thumbIconImageSource={thumbIconImageSource}
          thumbIconStyles={thumbIconStyles}
          title={title}
        />
      )}
    </View>
  );
};

SwipeButton.defaultProps = {
  containerStyles: {},
  disabled: false,
  disabledRailBackgroundColor: DISABLED_RAIL_BACKGROUND_COLOR,
  disabledThumbIconBackgroundColor: DISABLED_THUMB_ICON_BACKGROUND_COLOR,
  disabledThumbIconBorderColor: DISABLED_THUMB_ICON_BORDER_COLOR,
  height: 50,
  railBackgroundColor: RAIL_BACKGROUND_COLOR,
  railBorderColor: RAIL_BORDER_COLOR,
  railFillBackgroundColor: RAIL_FILL_BACKGROUND_COLOR,
  railFillBorderColor: RAIL_FILL_BORDER_COLOR,
  swipeSuccessThreshold: SWIPE_SUCCESS_THRESHOLD,
  thumbIconBackgroundColor: THUMB_ICON_BACKGROUND_COLOR,
  thumbIconBorderColor: THUMB_ICON_BORDER_COLOR,
  thumbIconStyles: {},
  title: 'Swipe to submit',
  titleColor: TITLE_COLOR,
  titleFontSize: 20,
  titleStyles: {},
};

SwipeButton.propTypes = {
  containerStyles: PropTypes.object,
  disable: PropTypes.bool,
  disabledRailBackgroundColor: PropTypes.string,
  disabledThumbIconBackgroundColor: PropTypes.string,
  disabledThumbIconBorderColor: PropTypes.string,
  enableRightToLeftSwipe: PropTypes.bool,
  forceReset: PropTypes.func,
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  onSwipeFail: PropTypes.func,
  onSwipeStart: PropTypes.func,
  onSwipeSuccess: PropTypes.func,
  railBackgroundColor: PropTypes.string,
  railBorderColor: PropTypes.string,
  railFillBackgroundColor: PropTypes.string,
  railFillBorderColor: PropTypes.string,
  railStyles: PropTypes.object,
  resetAfterSuccessAnimDelay: PropTypes.number,
  resetAfterSuccessAnimDuration: PropTypes.number,
  shouldResetAfterSuccess: PropTypes.bool,
  swipeSuccessThreshold: PropTypes.number, // Ex: 70. Swipping 70% will be considered as successful swipe
  thumbIconBackgroundColor: PropTypes.string,
  thumbIconBorderColor: PropTypes.string,
  thumbIconComponent: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.func,
  ]),
  thumbIconImageSource: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  thumbIconStyles: PropTypes.object,
  title: PropTypes.string,
  titleColor: PropTypes.string,
  titleFontSize: PropTypes.number,
  titleStyles: PropTypes.object,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

export default SwipeButton;
