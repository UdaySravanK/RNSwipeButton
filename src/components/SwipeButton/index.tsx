import React, { useState, useEffect, ReactElement, useCallback } from "react";
import {
  Text,
  AccessibilityInfo,
  TouchableOpacity,
  ViewStyle,
  TextStyle,
  ImageSourcePropType,
  LayoutChangeEvent,
  TouchableOpacityProps,
} from "react-native";

// Components
import SwipeThumb from "../SwipeThumb";

// Styles
import styles from "./styles";

// Constants
import {
  DEFAULT_ANIMATION_DURATION,
  DEFAULT_HEIGHT,
  DEFAULT_TITLE,
  DEFAULT_TITLE_FONT_SIZE,
  DEFAULT_TITLE_MAX_LINES,
  DISABLED_RAIL_BACKGROUND_COLOR,
  DISABLED_THUMB_ICON_BACKGROUND_COLOR,
  DISABLED_THUMB_ICON_BORDER_COLOR,
  RAIL_BACKGROUND_COLOR,
  RAIL_BORDER_COLOR,
  RAIL_FILL_BACKGROUND_COLOR,
  RAIL_FILL_BORDER_COLOR,
  SWIPE_SUCCESS_THRESHOLD,
  THUMB_ICON_BACKGROUND_COLOR,
  THUMB_ICON_BORDER_COLOR,
  TITLE_COLOR,
} from "../../constants";

interface SwipeButtonProps extends TouchableOpacityProps {
  containerStyles?: ViewStyle;
  disabled?: boolean;
  disabledRailBackgroundColor?: string;
  disabledThumbIconBackgroundColor?: string;
  disabledThumbIconBorderColor?: string;
  disableResetOnTap?: boolean;
  enableReverseSwipe?: boolean;
  finishRemainingSwipeAnimationDuration?: number;
  forceCompleteSwipe?: (forceComplete: () => void) => void;
  forceReset?: (forceReset: () => void) => void;
  height?: number;
  onSwipeFail?: () => void;
  onSwipeStart?: () => void;
  onSwipeSuccess?: (isForceComplete: boolean) => void;
  railBackgroundColor?: string;
  railBorderColor?: string;
  railFillBackgroundColor?: string;
  railFillBorderColor?: string;
  railStyles?: ViewStyle;
  resetAfterSuccessAnimDelay?: number;
  screenReaderEnabled?: boolean;
  shouldResetAfterSuccess?: boolean;
  swipeSuccessThreshold?: number;
  thumbIconBackgroundColor?: string;
  thumbIconBorderColor?: string;
  thumbIconComponent?: () => ReactElement;
  thumbIconImageSource?: ImageSourcePropType;
  thumbIconStyles?: ViewStyle;
  thumbIconWidth?: number;
  title?: string;
  titleColor?: string;
  titleFontSize?: number;
  titleMaxFontScale?: number;
  titleMaxLines?: number;
  titleStyles?: TextStyle;
  width?: number;
}

/**
 * - Height of the RNSwipeButton will be determines by the height of the inner ThumbIcon which we interact with to swipe.
 *
 * @param {*} param0
 * @returns
 */
const SwipeButton: React.FC<SwipeButtonProps> = ({
  containerStyles,
  disabled = false,
  disabledRailBackgroundColor = DISABLED_RAIL_BACKGROUND_COLOR,
  disabledThumbIconBackgroundColor = DISABLED_THUMB_ICON_BACKGROUND_COLOR,
  disabledThumbIconBorderColor = DISABLED_THUMB_ICON_BORDER_COLOR,
  disableResetOnTap = false,
  enableReverseSwipe,
  finishRemainingSwipeAnimationDuration = DEFAULT_ANIMATION_DURATION,
  forceCompleteSwipe,
  forceReset,
  height = DEFAULT_HEIGHT,
  onSwipeFail,
  onSwipeStart,
  onSwipeSuccess,
  railBackgroundColor = RAIL_BACKGROUND_COLOR,
  railBorderColor = RAIL_BORDER_COLOR,
  railFillBackgroundColor = RAIL_FILL_BACKGROUND_COLOR,
  railFillBorderColor = RAIL_FILL_BORDER_COLOR,
  railStyles,
  resetAfterSuccessAnimDelay,
  screenReaderEnabled,
  shouldResetAfterSuccess,
  swipeSuccessThreshold = SWIPE_SUCCESS_THRESHOLD,
  thumbIconBackgroundColor = THUMB_ICON_BACKGROUND_COLOR,
  thumbIconBorderColor = THUMB_ICON_BORDER_COLOR,
  thumbIconComponent,
  thumbIconImageSource,
  thumbIconStyles = {},
  thumbIconWidth,
  title = DEFAULT_TITLE,
  titleColor = TITLE_COLOR,
  titleFontSize = DEFAULT_TITLE_FONT_SIZE,
  titleMaxFontScale,
  titleMaxLines = DEFAULT_TITLE_MAX_LINES,
  titleStyles = {},
  width,
  ...rest // Include other TouchableOpacity props
}) => {
  const [layoutWidth, setLayoutWidth] = useState(0);
  const [isScreenReaderEnabled, setIsScreenReaderEnabled] =
    useState(screenReaderEnabled);
  const [isUnmounting, setIsUnmounting] = useState(false);
  const [activationMessage, setActivationMessage] = useState(title);
  const [disableInteraction, setDisableInteraction] = useState(false);
  /**
   * Retrieve layoutWidth to set maximum swipeable area.
   * Correct layout width will be received only after first render but we need it before render.
   * So render SwipeThumb only if layoutWidth > 0
   */
  const onLayoutContainer = useCallback(
    (e: LayoutChangeEvent) => {
      const newWidth = e.nativeEvent.layout.width;
      if (!isUnmounting && newWidth !== layoutWidth) {
        setLayoutWidth(newWidth);
      }
    },
    [isUnmounting, layoutWidth],
  );

  /**
   * If we don't update `disabled` prop of TouchableOpacity through state changes,
   * switching from a11y to normal mode would still keep the button in disabled state.
   * Which results to all interactions disabled. Swipe gesture won't work.
   */
  useEffect(() => {
    if (disabled && isScreenReaderEnabled) {
      setDisableInteraction(true);
    } else {
      setDisableInteraction(false);
    }
  }, [disabled, isScreenReaderEnabled]);

  const handleScreenReaderToggled = useCallback(
    (isEnabled: boolean) => {
      if (isUnmounting || isScreenReaderEnabled === isEnabled) {
        return;
      }
      if (screenReaderEnabled !== undefined) {
        setIsScreenReaderEnabled(screenReaderEnabled);
        // Return to avoid overriding the externally set value
        return;
      }

      setIsScreenReaderEnabled(isEnabled);
    },
    [isScreenReaderEnabled, screenReaderEnabled],
  );

  useEffect(() => {
    setIsUnmounting(false);
    const subscription = AccessibilityInfo.addEventListener(
      "screenReaderChanged",
      handleScreenReaderToggled,
    );

    AccessibilityInfo.isScreenReaderEnabled().then(handleScreenReaderToggled);

    return () => {
      setIsUnmounting(true);
      if (subscription) {
        subscription.remove();
      }
    };
  }, [isScreenReaderEnabled, handleScreenReaderToggled]);

  useEffect(() => {
    // Update activation message based on disabled state and screen reader status
    if (disabled) {
      setActivationMessage("Button disabled");
    } else if (isScreenReaderEnabled) {
      setActivationMessage("Double tap to activate");
    } else {
      setActivationMessage(title);
    }
  }, [disabled, isScreenReaderEnabled]);

  const handlePress = useCallback(() => {
    if (disabled) return;
    if (isScreenReaderEnabled) {
      // Simulate swipe success for screen readers
      onSwipeSuccess && onSwipeSuccess(false);
    }
  }, [disabled, isScreenReaderEnabled, onSwipeSuccess]);

  const handleFocus = useCallback(() => {
    AccessibilityInfo.isScreenReaderEnabled().then(handleScreenReaderToggled);
  }, [handleScreenReaderToggled]);

  const dynamicContainerStyles: ViewStyle = {
    ...containerStyles,
    backgroundColor: disabled
      ? disabledRailBackgroundColor
      : railBackgroundColor,
    borderColor: railBorderColor,
    ...(width ? { width } : {}),
  };

  return (
    <TouchableOpacity
      onFocus={handleFocus}
      disabled={disableInteraction}
      accessible={true}
      accessibilityLabel={title}
      accessibilityHint={activationMessage}
      accessibilityRole="button"
      activeOpacity={disabled ? 1 : 0.5} // Make it feel like a button
      onPress={handlePress}
      style={[styles.container, { ...dynamicContainerStyles }]}
      onLayout={onLayoutContainer}
      testID="SwipeButton"
      {...rest}
    >
      <Text
        maxFontSizeMultiplier={titleMaxFontScale}
        ellipsizeMode={"tail"}
        numberOfLines={titleMaxLines}
        style={[
          styles.title,
          {
            color: titleColor,
            fontSize: titleFontSize,
            ...titleStyles,
          },
        ]}
      >
        {title}
      </Text>
      {layoutWidth > 0 && (
        <SwipeThumb
          disabled={disabled}
          disabledThumbIconBackgroundColor={disabledThumbIconBackgroundColor}
          disabledThumbIconBorderColor={disabledThumbIconBorderColor}
          disableResetOnTap={disableResetOnTap}
          enableReverseSwipe={enableReverseSwipe}
          finishRemainingSwipeAnimationDuration={
            finishRemainingSwipeAnimationDuration
          }
          forceCompleteSwipe={forceCompleteSwipe}
          forceReset={forceReset}
          layoutWidth={layoutWidth}
          onSwipeFail={onSwipeFail}
          onSwipeStart={onSwipeStart}
          onSwipeSuccess={onSwipeSuccess}
          railFillBackgroundColor={railFillBackgroundColor}
          railFillBorderColor={railFillBorderColor}
          railStyles={railStyles}
          resetAfterSuccessAnimDelay={resetAfterSuccessAnimDelay}
          shouldResetAfterSuccess={shouldResetAfterSuccess}
          swipeSuccessThreshold={swipeSuccessThreshold}
          thumbIconBackgroundColor={thumbIconBackgroundColor}
          thumbIconBorderColor={thumbIconBorderColor}
          thumbIconComponent={thumbIconComponent}
          thumbIconHeight={height}
          thumbIconImageSource={thumbIconImageSource}
          thumbIconStyles={thumbIconStyles}
          thumbIconWidth={thumbIconWidth}
        />
      )}
    </TouchableOpacity>
  );
};

export default React.memo(SwipeButton);
