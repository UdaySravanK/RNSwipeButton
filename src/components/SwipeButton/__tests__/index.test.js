import { render } from "@testing-library/react-native";

import SwipeButton from "../index";
import { expect } from "@jest/globals";

describe("Component: SwipeButton", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly with default props", async () => {
    const { toJSON } = render(<SwipeButton />);
    expect(toJSON()).toMatchSnapshot();
  });
});
