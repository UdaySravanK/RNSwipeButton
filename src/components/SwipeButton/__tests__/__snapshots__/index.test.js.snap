// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SwipeButton should render correctly with default props 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "borderRadius": 50,
        "borderWidth": 1,
        "justifyContent": "center",
        "margin": 5,
      },
      {
        "backgroundColor": "#7CF3F9",
        "borderColor": "#073436",
      },
    ]
  }
  testID="SwipeButton"
>
  <Text
    ellipsizeMode="tail"
    importantForAccessibility=""
    numberOfLines={1}
    style={
      [
        {
          "alignSelf": "center",
          "position": "absolute",
        },
        {
          "color": "#083133",
          "fontSize": 20,
        },
      ]
    }
  >
    Swipe to submit
  </Text>
</View>
`;
